File: src/main/java/com/matag/game/deck/DeckRetrieverService.java
package com.matag.game.deck;

import org.springframework.stereotype.Component;

import com.matag.adminentities.DeckInfo;
import com.matag.game.adminclient.AdminClient;
import com.matag.game.security.SecurityToken;

import lombok.AllArgsConstructor;

@Component
@AllArgsConstructor
public class DeckRetrieverService {
  private final AdminClient adminClient;

  public DeckInfo retrieveDeckForUser(SecurityToken token) {
    return adminClient.getDeckInfo(token);
  }
}

File: src/main/kotlin/com/matag/game/deck/DeckRetrieverService.kt
package com.matag.game.deck

import com.matag.adminentities.DeckInfo
import com.matag.game.adminclient.AdminClient
import com.matag.game.security.SecurityToken
import org.springframework.stereotype.Component

@Component
class DeckRetrieverService(
    private val adminClient: AdminClient
) {
    fun retrieveDeckForUser(token: SecurityToken): DeckInfo =
        adminClient.getDeckInfo(token)
}
